src/modules/landing_target_estimator/LandingTargetEstimator.cpp

```diff
@@ -256,28 +248,76 @@ void LandingTargetEstimator::_update_topics()
 
 		_new_sensorReport = true;
 
-	} else if (_uwbDistanceSub.update(&_uwbDistance)) {
+	} else if (_sensorUwbSub.update(&_sensorUwb)) {
+
 		if (!_vehicleAttitude_valid || !_vehicleLocalPosition_valid) {
 			// don't have the data needed for an update
 			PX4_INFO("Attitude: %d, Local pos: %d", _vehicleAttitude_valid, _vehicleLocalPosition_valid);
 			return;
 		}
 
-		if (!matrix::Vector3f(_uwbDistance.position).isAllFinite()) {
-			PX4_WARN("Position is corrupt!");
+		if (!PX4_ISFINITE(_sensorUwb.distance)) {
+			PX4_WARN("Data is corrupt!");
 			return;
 		}
 
-		_new_sensorReport = true;
+		// First we need to catch angle measurements outside of the useable measuring range
+		if ((float)(60.0) <= _sensorUwb.aoa_azimuth_dev || (float)(-60.0) >= _sensorUwb.aoa_azimuth_dev) {
+			return;
+		}
+
+		if ((float) 60.0 <= _sensorUwb.aoa_elevation_dev  || (float) -60.0 >= _sensorUwb.aoa_elevation_dev) {
+			return;
+		}
 
-		// The coordinate system is NED (north-east-down)
-		// the uwb_distance msg contains the Position in NED, Vehicle relative to LP
-		// The coordinates "rel_pos_*" are the position of the landing point relative to the vehicle.
-		// To change POV we negate every Axis:
-		_target_position_report.timestamp = _uwbDistance.timestamp;
-		_target_position_report.rel_pos_x = -_uwbDistance.position[0];
-		_target_position_report.rel_pos_y = -_uwbDistance.position[1];
-		_target_position_report.rel_pos_z = -_uwbDistance.position[2];
+		_new_sensorReport = true;
+		_target_position_report.timestamp = _sensorUwb.timestamp;
+
+		const float deg2rad = M_PI / 180.0;
+		float azimuth 	 = _sensorUwb.aoa_azimuth_dev *  deg2rad; 	//subtract yaw offset and convert to rad
+		float elevation = _sensorUwb.aoa_elevation_dev  * deg2rad; 	//subtract pitch offset and convert to rad
```

Comment:
Can be made `const`

```suggestion
		const float azimuth 	 = _sensorUwb.aoa_azimuth_dev *  deg2rad; 	//subtract yaw offset and convert to rad
		const float elevation = _sensorUwb.aoa_elevation_dev  * deg2rad; 	//subtract pitch offset and convert to rad
```

Reply:
Can not be constant as it _sensorUwb.aoa_elevation_dev changes with each measurement

Reply:
But not inside this function scope, or am I blind? These variables are defined here and then used twice as read-only variables right after the comment below.

Reply:
`matrix::Vector3f position = - matrix::Vector3f{(_sensorUwb.distance  * sinf(math::radians(_sensorUwb.aoa_azimuth_dev)) * cosf(math::radians(_sensorUwb.aoa_elevation_dev))),
				 (_sensorUwb.distance  * sinf(math::radians(_sensorUwb.aoa_elevation_dev))),
				 (_sensorUwb.distance * cosf(math::radians(_sensorUwb.aoa_azimuth_dev)) * cosf(math::radians(_sensorUwb.aoa_elevation_dev)))};`
				 
using const would prevent the value to be updated when a new sensor_uwb message is published right?